worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -319,7 +319,8 @@ function processFailures(document) {
                     COMPONENT: source.trim() + " " + numericVersion.trim(),
                     WORKFLOW_NAME: document.getRootDocument().getField("CAF_WORKFLOW_NAME").getStringValues().get(0),
                     MESSAGE: f.getFailureMessage(),
-                    DATE: new Date().toISOString()
+                    DATE: new Date().toISOString(),
+                    CORRELATION_ID: document.getRootDocument().getCustomData("correlationId") || undefined
```

Comment:
Change to:

    CORRELATION_ID: document.getCustomData("correlationId") || undefined

Custom data is a task level thing - it will not be different between the different documents in the task.